lib/spack/spack/version.py

```diff
@@ -348,11 +348,25 @@ def isdevelop(self):
         return False
 
     @coerced
-    def satisfies(self, other):
-        """A Version 'satisfies' another if it is at least as specific and has
-        a common prefix.  e.g., we want gcc@4.7.3 to satisfy a request for
-        gcc@4.7 so that when a user asks to build with gcc@4.7, we can find
-        a suitable compiler.
+    def intersects(self, other: "VersionBase") -> bool:
+        """Return True if self intersects with other, False otherwise.
+
+        Two versions intersect if one can be constrained by the other. For instance
+        @4.7 and @4.7.3 intersect (the intersection being @4.7.3).
+
+        Arg:
+            other: version to be checked for intersection
+        """
+        return self.satisfies(other) or other.satisfies(self)
```

Comment:
Hm yeah, so this works because version @x interpreted as @x:x means that @x and @y have overlap if and only if one is contained in the other... not very pretty

Reply:
I would implement it as 

```python
n = min(len(self.version), len(other.version))
return self.version[:n] == other.version[:n]
```

Reply:
Done in https://github.com/spack/spack/pull/35681/commits/76d4ef974669290b9d8026f8bb4fbf29c292dff2